website/server/models/task.js

```diff
@@ -25,6 +26,13 @@ export let TaskSchema = new Schema({
   type: {type: String, enum: tasksTypes, required: true, default: tasksTypes[0]},
   text: {type: String, required: true},
   notes: {type: String, default: ''},
+  shortName: {
+    type: String,
+    validate: [function validateShortName (val) {
+      return Boolean(this.userId) && !validator.isUUID(val) && val.match(SHORT_NAME_VALIDATION_REGEX);
+    },
+    'Task short names can only contain alphanumeric characters, underscores and dashses. They must be owned by a user.'],
```

Comment:
Oh! I think it's possible to add multiple validators for mongoose properties. Going to look into that, and that should help make the error messages more helpful. 

And yes, a challenge owner cannot set a short name on a task. And, in the future, if we set up tasks shared by groups, short names will not be set for those. IE, it requires a user to have a short name.